src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -148,6 +168,39 @@ private static void CheckIsEmpty<T>(T x)
         }
     }
 
+    /// <summary>
+    /// Context passed to a source generator when it has opted-in to PostInitialization via <see cref="GeneratorInitializationContext.RegisterForPostInitialization(Action{GeneratorPostInitializationContext})"/>
+    /// </summary>
+    public readonly struct GeneratorPostInitializationContext
```

Comment:
Should this provide access to a compilation and/or msbuild info?

Reply:
@333fred I mentioned this in the PR description, but we're deliberately keeping it simple for now. We can add more to the struct down the line if/when we have explicit customer scenarios that require it.